scipy/_lib/_array_api.py

```diff
@@ -545,3 +545,25 @@ def xp_sign(x: Array, /, *, xp: ModuleType | None = None) -> Array:
     sign = xp.where(x < 0, -one, sign)
     sign = xp.where(x == 0, 0*one, sign)
     return sign
+
+# maybe use `scipy.linalg` if/when array API support is added
+def xp_vector_norm(x: Array, /, *, 
+                   axis: int | tuple[int] | None = None,
+                   keepdims: bool = False,
+                   ord: int | float = 2,
+                   xp: ModuleType | None = None) -> Array:
+    xp = array_namespace(x) if xp is None else xp
+
+    # check for optional `linalg` extension
+    if SCIPY_ARRAY_API and hasattr(xp, 'linalg'):
+        return xp.linalg.vector_norm(x, axis=axis, keepdims=keepdims, ord=ord)
+
+    else:
+        if ord != 2:
+            raise ValueError(
+                "only the Euclidean norm (`ord=2`) is currently supported in "
+                "`xp_vector_norm` for backends not implementing the `linalg` extension."
+            )
+        # return (x @ x)**0.5 
+        # or to get the right behavior with nd, complex arrays
+        return xp.sum(xp.conj(x) * x, axis=axis, keepdims=keepdims)**0.5
```

Comment:
I don't quite fully understand this. If `SCIPY_ARRAY_API=0` then `xp.sum(xp.conj(x) * x, axis=axis, keepdims=keepdims)**0.5` is always used which doesn't seem preferable compared to `np.linalg.norm`?

Reply:
Right, I think I should rework the logic to

```
if SCIPY_ARRAY_API:
    if hasattr(xp, linalg):
        ...
    else:
        ...
else:
    # use np.linalg.norm
```

Reply:
this should be fixed now